packages/web/src/common/store/queue/sagas.ts

```diff
@@ -181,13 +187,16 @@ export function* watchPlay() {
       const noTrackPlaying = !playerTrackId
       const trackIsDifferent = playerTrackId !== playActionTrack.track_id
       const trackIsSameButDifferentUid =
-        playerTrackId === playActionTrack.track_id && uid !== playerUid
+        playerTrackId === playActionTrack.track_id &&
+        (uid !== playerUid || !!isPreview !== playerIsPreviewing)
       if (noTrackPlaying || trackIsDifferent || trackIsSameButDifferentUid) {
         yield* put(
           playerActions.play({
             uid,
+            isPreview,
             trackId: playActionTrack.track_id,
-            onEnd: next
+            // Don't auto-advance after previews
+            onEnd: isPreview ? playerActions.stop : next
```

Comment:
Are you sure that's what we want? In lineups, that would make streaming stop, no?

Reply:
I can follow up with design and double check. Generally I think we need to clarify the subtleties of how previewable tracks interact with lineups and queu.